docs/user-guide/cluster-docker-consul.md

```diff
@@ -0,0 +1,278 @@
+
+Why we need Traefik in cluster mode? Running multiple instances should work out of the box?
+
+If you want to use Let's Encrypt with Traefik, sharing configuration or TLS certificates between many Træfik instances, you need Traefik cluster/HA.
+
+Ok, could we mount a shared volume used by all my instances? Yes, you can, but it will not work.
+When you use Let's Encrypt, you need to store certificates, but not only.
+When Traefik generates a new certificate, it configures a challenge and once Let's Encrypt will verify the ownership of the domain, it will ping back the challenge. If the challenge is not knowing by other Traefik instances, the validation will fail.
+
+For more information about challenge: [Automatic Certificate Management Environment (ACME)](https://github.com/ietf-wg-acme/acme/blob/master/draft-ietf-acme-acme.md#tls-with-server-name-indication-tls-sni)
+
+## Prerequisites
+
+You will need a working Docker Swarm cluster.
+
+## Traefik configuration
+
+In this guide, we will not use a toml configuration file, but only command line flag.
+With that, we can use the base image without mouting configuration file or building custom image.
+
+What Traefik should do:
+- Listen to 80 and 443
+- Redirect HTTP traffic to HTTPS
+- Generate SSL certificate when a domain is added
+- Listen to Docker Swarm event
+
+### Entrypoints configuration
+
+TL;DR:
+```
+$ traefik \
+    --entrypoints=Name:http Address::80 Redirect.EntryPoint:https \
+    --entrypoints=Name:https Address::443 TLS \
+    --defaultentrypoints=http,https
+```
+
+To listen to different ports, we need to create an entrypoint for each.
+The CLI syntaxe is `--entrypoints=Name:a_name Address:an_ip_or_empty:a_port options`.
+If you want to redirect traffic from one entrypoint to another, it's the option `Redirect.EntryPoint:entrypoint_name`.
+
+By default, we don't want to configure all our services to listen on http and https, we add a default entrypoint configuration: `--defaultentrypoints=http,https`.
+
+### Let's Encrypt configuration
+
+TL;DR:
+```
+$ traefik \
+    --acme \
+    --acme.storage=/etc/traefik/acme/acme.json \
+    --acme.entryPoint=https \
+    --acme.email=contact@mydomain.ca
+```
+
+Let's Encrypt needs 3 parameters: an entryPoint to listen to, a storage for certificates, and en email for the registration.
+
+To enable Let's Encrypt support, you need to add `--acme` flag.
+
+Now, Traefik needs to know where to store the certificates, we can choose between a key in a Key-Value store, or a file path: `--acme.storage=my/key` or `--acme.storage=/path/to/acme.json`.
+
+For your email and the entrypoints, it's `--acme.entryPoint` and `--acme.email` flags.
+
+### Docker configuration
+
+TL;DR:
+```
+$ traefik \
+    --docker \
+    --docker.swarmmode \
+    --docker.domain=mydomain.ca \
+    --docker.watch
+```
+To enable docker and support, you need to add `--docker` and `--docker.swarmmode` flags.
+To enable watch docker changes, add `--docker.watch`.
```

Comment:
WDYT about simplying the sentence : **To watch docker changes** ?

Reply:
WDYT of: *To watch docker events* ?